ansible/playbooks/cluster-prepare.yml

```
@@ -65,12 +65,17 @@
 
     - name: System Configuration (1)
       block:
-        - name: System Configuration (1) | Disable ufw
+        - name: System Configuration (1) | Check if ufw is installed/running
+          ansible.builtin.service_facts:
+        - name: System Configuration (1) | Disable ufw if needed
           ansible.builtin.systemd:
             service: ufw.service
             enabled: false
             masked: true
             state: stopped
+          when:
+            - "'ufw' in ansible_facts.services"
+            - ansible_facts.services['ufw']['state'] == "running"
```

I kind of like this method from stackoverflow better: https://stackoverflow.com/a/73646230/10496442

```suggestion
        - name: System Configuration (1) | Disable ufw
          ansible.builtin.systemd:
            service: ufw.service
            enabled: false
            masked: true
            state: stopped
		  register: stop_service
		  failed_when:
		    - stop_service.failed == true
		    - '"Could not find the requested service" not in stop_service.msg'
```

---

Agreed, thanks!